pkg/volume/cinder/cinder_util.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cinder
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/cloudprovider/openstack"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/mount"
+	"github.com/golang/glog"
+)
+
+type CinderDiskUtil struct{}
+
+// Attaches a disk specified by a volume.CinderPersistenDisk to the current kubelet.
+// Mounts the disk to it's global path.
+func (util *CinderDiskUtil) AttachDisk(cd *cinderVolume, globalPDPath string) error {
+
+	flags := uintptr(0)
+	if cd.readOnly {
+		flags = mount.FlagReadOnly
+	}
+	cloud := cd.plugin.host.GetCloudProvider()
```

Comment:
I'm torn on this one - this is probably a better answer but I want to do it consistently across the various forms, and I am not sure if this is 100% there (just haven't thought deeply).  Shortest path to merge is to do it the same way others do and then file a bug to fix them all.